Cargo.toml

```diff
@@ -34,6 +34,7 @@ xdg = "2.2"
 librespot = { version = "0.1.1", default-features = false, features = ["with-tremor"] }
 toml = "0.5.8"
 color-eyre = "0.5"
+cfg-if = "1"
```

Comment:
im usually a bit hesitant about adding a dependency, and `cfg_ifs` main use case seems to be long `cfg` chains, while we really only have `#[cfg(feature="alsa")]` and  `#[not(cfg(feature="alsa"))]` so im in favor of doing it manually. opinions @Spotifyd/maintainers ?

Reply:
I can understand the `cfg` stuff. At the end looks easier to express conditional internal fn code with this macro rather than with normal `cfg`s IMO.

The main benefit I've always found from using `cfg-if` aside from the syntax it provides (which is more clear IMO) is that you end up with less code normally. Which is always good.

Happy to hear your opinions and go on the direction you decide!

Reply:
> #[not(cfg(feature="alsa"))]

`#cfg[not(feature = "alsa")]`? :thinking:

Reply:
@SirWindfield haha yea that seems to be the correct one, do you have an opinion here? 

@CPerezz have you used it before?

thinking about it, it might be nice to put all feature cfg logic outside of the main files, but that not really relevant to this PR

Reply:
There's a problem with @SirWindfield approach if I'm not mistaken (I've never tried) and it's that IIRC `#cfg[not(feature = "alsa")]` cannot be used inside of fn definition code. That's why `cfg-if` exists. It dessugars into the `#cfg[not(feature = "alsa")] and #cfg[(feature = "alsa")]` fn definitions. But you write a way less code and it's easier to read.

So I'm not really sure that the way on which you want to implement it will work considering that I used `#cfg[not(feature = "alsa")]` flagging style everywhere possible (inside structs or similar things).

Reply:
I really do not care to be honest. If it reduces boilerplate, go for it. If it is just sugar that removes two lines of code, I'd rather have those lines written.

To give a more specific example: if `cfg-if` allows to remove a whole "duplicate" function body I'd take the trade-off, if it's just a line of code that can properly be filtered I'd rather work without it.